lib/matplotlib/tests/test_agg.py

```diff
@@ -155,6 +159,140 @@ def test_long_path():
+        y = np.convolve(w/w.sum(), s, mode='same')
+        return y[window_len-1:-window_len+1]
+
+    def smooth2d(A, sigma=3):
+        window_len = max(int(sigma), 3)*2 + 1
+        A1 = np.array([smooth1d(x, window_len) for x in np.asarray(A)])
+        A2 = np.transpose(A1)
+        A3 = np.array([smooth1d(x, window_len) for x in A2])
+        A4 = np.transpose(A3)
+
+        return A4
+
+    class BaseFilter(object):
+        def prepare_image(self, src_image, dpi, pad):
+            ny, nx, depth = src_image.shape
+            padded_src = np.zeros([pad*2 + ny, pad*2 + nx, depth], dtype="d")
+            padded_src[pad:-pad, pad:-pad, :] = src_image[:, :, :]
+
+            return padded_src  # , tgt_image
+
+        def get_pad(self, dpi):
+            return 0
+
+        def __call__(self, im, dpi):
+            pad = self.get_pad(dpi)
+            padded_src = self.prepare_image(im, dpi, pad)
+            tgt_image = self.process_image(padded_src, dpi)
+            return tgt_image, -pad, -pad
+
+    class OffsetFilter(BaseFilter):
+        def __init__(self, offsets=None):
+            if offsets is None:
+                self.offsets = (0, 0)
+            else:
+                self.offsets = offsets
+
+        def get_pad(self, dpi):
+            return int(max(*self.offsets)/72.*dpi)
+
+        def process_image(self, padded_src, dpi):
+            ox, oy = self.offsets
+            a1 = np.roll(padded_src, int(ox/72.*dpi), axis=1)
+            a2 = np.roll(a1, -int(oy/72.*dpi), axis=0)
+            return a2
+
+    class GaussianFilter(BaseFilter):
+        "simple gauss filter"
+
+        def __init__(self, sigma, alpha=0.5, color=None):
+            self.sigma = sigma
+            self.alpha = alpha
+            if color is None:
+                self.color = (0, 0, 0)
+            else:
+                self.color = color
+
+        def get_pad(self, dpi):
+            return int(self.sigma*3/72.*dpi)
+
+        def process_image(self, padded_src, dpi):
+            #offsetx, offsety = int(self.offsets[0]), int(self.offsets[1])
+            tgt_image = np.zeros_like(padded_src)
+            aa = smooth2d(padded_src[:, :, -1]*self.alpha,
+                          self.sigma/72.*dpi)
+            tgt_image[:, :, -1] = aa
+            tgt_image[:, :, :-1] = self.color
+            return tgt_image
+
+    class DropShadowFilter(BaseFilter):
+        def __init__(self, sigma, alpha=0.3, color=None, offsets=None):
+            self.gauss_filter = GaussianFilter(sigma, alpha, color)
+            self.offset_filter = OffsetFilter(offsets)
+
+        def get_pad(self, dpi):
+            return max(self.gauss_filter.get_pad(dpi),
+                       self.offset_filter.get_pad(dpi))
+
+        def process_image(self, padded_src, dpi):
+            t1 = self.gauss_filter.process_image(padded_src, dpi)
+            t2 = self.offset_filter.process_image(t1, dpi)
+            return t2
+
+    if V(np.__version__) <= V('1.7.0'):
```

Comment:
The use of `<=` is just a mistake.  Will fix.

I used an `if` here because `@knownfailureif` and `@skiptest` interact poorly with `@image_comparison`.  We should just fix that at some point, but I think that's out-of-scope for this PR.  (There are other tests already that have to do this).